cpr/connection_pool.cpp

```diff
@@ -0,0 +1,39 @@
+#include "cpr/connection_pool.h"
+#include <curl/curl.h>
+#include <memory>
+#include <mutex>
+
+namespace cpr {
+ConnectionPool::ConnectionPool() {
+    auto* curl_share = curl_share_init();
```

Comment:
```suggestion
    const CURLSH* curl_share = curl_share_init();
```
I'm trying to avoid `auto` when ever possible since I prefer explicit code.